rom/dev/src/fuse.rs

```diff
@@ -0,0 +1,68 @@
+/*++
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    fuse.rs
+
+Abstract:
+
+    The file contains Fuse-related Implementations.
+
+--*/
+use caliptra_common::{FuseLogEntry, FuseLogEntryId};
+use caliptra_drivers::CaliptraResult;
+use caliptra_error::caliptra_err_def;
+use zerocopy::AsBytes;
+
+caliptra_err_def! {
+    RomGlobal,
+    GlobalErr
+    {
+        FuseLogInvalidEntryId = 0x6,
+        FuseLogUnsupportedDataLength = 0x7,
+    }
+}
+
+extern "C" {
+    static mut FUSE_LOG_ORG: u8;
+}
+
+/// Log Fuse data
+///
+/// # Arguments
+/// * `entry_id` - log entry ID
+/// * `data` -  data To log to the fuse log
+///
+/// # Return Value
+/// * `Ok(())` - Success
+/// * `Err(GlobalErr::FuseLogInvalidEntryId)` - Invalid Fuse log entry ID
+/// * `Err(GlobalErr::FuseLogUpsupportedDataLength)` - Unsupported data length
+///
+pub fn log_fuse_data(entry_id: FuseLogEntryId, data: &[u8]) -> CaliptraResult<()> {
+    if entry_id == FuseLogEntryId::Invalid {
+        raise_err!(FuseLogInvalidEntryId);
+    }
+
+    if data.len() > 8 {
+        raise_err!(FuseLogUnsupportedDataLength);
+    }
+
+    // Create a FUSE log entry
+    let mut log_entry = FuseLogEntry {
+        entry_id: entry_id as u32,
+        ..Default::default()
+    };
+    log_entry.log_data.as_bytes_mut()[..data.len()].copy_from_slice(data);
+
+    let dst = unsafe {
+        let offset = core::mem::size_of::<FuseLogEntry>() * (entry_id as usize - 1);
+        let ptr = (&mut FUSE_LOG_ORG as *mut u8).add(offset);
```

Comment:
Instead of using this pattern, I would recommend something like

```
extern "C" {
    static mut FUSE_LOG_ORG: [FuseLogEntry; FUSE_LOG_SIZE / size_of::<FuseLogEntry>()];
}
```

That way you can write code that directly indexes that slice, do bounds checks, etc.

Reply:
Updated the code.